app/src/utils/MadLibs.ts

```diff
@@ -8,18 +8,21 @@ export type PhraseSelections = Record<number, number>;
 export type PhraseSegment = string | Record<number, string>;
 
 export interface MadLib {
+  readonly index: number;
   readonly phrase: PhraseSegment[];
   readonly defaultSelections: PhraseSelections;
+  readonly activeSelections: PhraseSelections;
 }
 
-function getMadLibPhraseText(
-  madLib: MadLib,
-  phraseSelections: PhraseSelections
-): string {
+function getMadLibPhraseText(madLib: MadLib): string {
   let madLibText = "";
   madLib.phrase.forEach((phraseSegment, index) => {
     if (phraseSegment.constructor === Object) {
-      madLibText += " " + phraseSegment[phraseSelections[index]] + " ";
+      let selection =
+        madLib.activeSelections && madLib.activeSelections[index]
```

Comment:
This can't ever be null/undefined, can it?

Reply:
oh yeah! i had it optional at one point and then decided against it